src/SimulationRunner_TEST.cc

```diff
@@ -1135,6 +1138,138 @@ TEST_P(SimulationRunnerTest, LoadPlugins)
   #endif
 }
 
+/////////////////////////////////////////////////
+TEST_P(SimulationRunnerTest, LoadServerConfigPlugins)
+{
+  sdf::Root rootWithout;
+  rootWithout.Load(std::string(PROJECT_SOURCE_PATH) +
+      "/test/worlds/plugins_empty.sdf");
+  ASSERT_EQ(1u, rootWithout.WorldCount());
+
+  std::string plugins = R"(
+  <server_config>
+    <plugin
+      entity_name="default"
+      entity_type="world"
+      filename="libTestWorldSystem.so"
+      name="ignition::gazebo::TestWorldSystem">
+      <world_key>0.123</world_key>
+    </plugin>
+    <plugin
+      entity_name="box"
+      entity_type="model"
+      filename="libTestModelSystem.so"
+      name="ignition::gazebo::TestModelSystem">
+      <model_key>987</model_key>
+    </plugin>
+    <plugin
+      entity_name="default::box::link_1::camera"
+      entity_type="sensor"
+      filename="libTestSensorSystem.so"
+      name="ignition::gazebo::TestSensorSystem">
+      <sensor_key>456</sensor_key>
+    </plugin>
+  </server_config>)";
+
+  // Create a server configuration with plugins
+  ServerConfig serverConfig;
+  for (auto plugin : ParsePluginsFromString(plugins))
```

Comment:
Can we also have a test for `ParsePluginsFromFile`?

Reply:
Added in a separate `ServerConfig_TEST.cc`